run.ts

```diff
@@ -63,37 +63,32 @@ const launchDocker: SearchEngineLauncherFunction = async ({
   port,
   options,
 }) => {
-  const Image =
-    engine === 'elasticsearch'
-      ? 'elastic/elasticsearch:8.6.2'
-      : 'opensearchproject/opensearch:2.11.0'
-  console.log('Launching Docker container', Image)
-  const docker = new Dockerode()
-  const container = await docker.createContainer({
-    Env: [...options, 'path.data=/var/lib/search', 'path.logs=/var/log/search'],
-    HostConfig: {
-      AutoRemove: true,
-      Mounts: [
-        { Source: dataDir, Target: '/var/lib/search', Type: 'bind' },
-        { Source: logsDir, Target: '/var/log/search', Type: 'bind' },
-      ],
-      PortBindings: {
-        [`${port}/tcp`]: [{ HostIP: '127.0.0.1', HostPort: `${port}` }],
-      },
-    },
-    Image,
+  const argv = {
+    dataDir,
+    logsDir,
+    engine,
+    port,
+    options,
+  }
+  const subprocess = fork(
+    './node_modules/@nasa-gcn/architect-plugin-search/launchSearch.js',
```

Comment:
I'd rather not hardcode this path. Can you `__filename` like in the example code in https://nodejs.org/api/child_process.html#child_processforkmodulepath-args-options?

Reply:
There are 3 main ways to get where a file is located. __dirname to get the directory, __filename to get the file, or process.cwd() to get the current working directory of the process. I had to do a little bit of fiddling to get __dirname and __filename to work:
<img width="744" alt="Screenshot 2024-06-12 at 1 24 52 PM" src="https://github.com/nasa-gcn/architect-plugin-search/assets/6545874/0b418bc3-fd9d-42b2-8e3a-8bfed2052182">

As long as the package is not symlinked it works as expected:
<img width="702" alt="Screenshot 2024-06-12 at 1 37 54 PM" src="https://github.com/nasa-gcn/architect-plugin-search/assets/6545874/049503de-cb25-4558-a458-8f1b236328bf">

however, if you symlink the package to work locally, it will not work:
<img width="446" alt="Screenshot 2024-06-12 at 1 23 36 PM" src="https://github.com/nasa-gcn/architect-plugin-search/assets/6545874/5063bbfd-d2fb-404b-bad8-d5c924973dff">
The problem is that it points to the path where of code that you are symlinking, not the copy in .node_modules (even though it does exist in .node_modules, which is why hard coding worked)

I am not sure how often not being able to symlink is a dealbreaker for local development. I use it, but I don't know if it's widely used by others using the package. 

Would you still like me to use __dirname + the file name?

Reply:
I added code in the last commit that checks to see if the directory includes node_modules in the path. if it does it just uses __dirname but if it doesn't (which would mean a symlink) then it returns the hard coded path to the node_modules file. 
It's not too much cleaner than just hard coding it, but a little better.

Reply:
Take another look at the sample code fork `child_process.fork` in the Node.js documentation:

```js
if (process.argv[2] === 'child') {
  setTimeout(() => {
    console.log(`Hello from ${process.argv[2]}!`);
  }, 1_000);
} else {
  const { fork } = require('node:child_process');
  const controller = new AbortController();
  const { signal } = controller;
  const child = fork(__filename, ['child'], { signal });
  child.on('error', (err) => {
    // This will be called with err being an AbortError if the controller aborts
  });
  controller.abort(); // Stops the child process
}
```

It's the same file. You have some variable --- in this case `process.argv` --- that you use to check if you are in the parent process or the child process.

This is analogous to the typical usage of the `fork()` syscall. See [example](https://man7.org/linux/man-pages/man2/fork.2.html#EXAMPLES).

Reply:
I'm not sure what you are saying exactly here.
It sounds like you are giving me advice on how to tell what is the parent vs child process. This is not an issue that I am having. 

The issue that you initially pointed out was that the filename was hard coded. You wanted me to use something similar to __filename. (__filename doesn't work out of the box in es6 so you have to do a few extra steps to define it.)

The issue I encountered is that if have symlinked architect-plugin-search in another application for development, the path to the launchSearch.js file is different than if it is not symlinked. When symlinked the path to the file points to my local  architect-plugin-search repo which is a different path than when it is in node_modules. 

Presumably the two are in sync since they are a simlink, so whatever changes you make in the imported package from the node_modules directory will be reflected in your local repo. Which makes this not really a big deal (it will run either way). It just is weird that the path would point to the linked repo instead of the node_module within the application. So my solution is just to define __dirname and leave it at that. It can run in the symlinked repo or it can run in the node_modules if not symlinked. Either way works. 

So I just ended up removing the check to see if node_modules is in the path. It'll work either way, it'll just be pointing to the external repo instead of the node_modules if symlinked.

Reply:
I understand now that `__filename` only works in CommonJS modules. Just use `import.meta.url` instead. No need to convert it to a path, because `child_process.fork` accepts URLs.

Why is it a problem that this might return the target of the symlink?

Reply:
import.meta.url is `architect-plugin-search/index.js` so that can't be used directly. Here is what I'm doing there and why it needs to have multiple steps:

1. import.meta.url: `file:///Users/courey/dev/architect-plugin-search/index.js` this is getting the name of the file that is currently running. in this case it's `index.js`. I don't want `index.js`, I want `launchSearch.js`.
2. so we then pass that into `fileURLToPath` so __filename would then be: `/Users/courey/dev/architect-plugin-search/index.js`. This is preparing it so that we can use path. 
3. I don't want the file, just the directory. Since we converted that to a path with `fileURLToPath` we can use `path.dirname()` to get just the path of the directory instead of the file that is running. that makes __dirname: `/Users/courey/dev/architect-plugin-search`
4. I add `/launchSearch.js` to the end of that path to have the proper path and proper file name.

## stuff that doesn't work

If I just used `import.meta.url`  we get this error:
```
Error: Cannot find module '/Users/courey/dev/gcn.nasa.gov/file:/Users/courey/dev/architect-plugin-search/index.js'
```

if I added `/launchSearch.js` to the `import.meta.url` we get: 
```Error: Cannot find module '/Users/courey/dev/gcn.nasa.gov/file:/Users/courey/dev/architect-plugin-search/index.js/```

import.meta.url passed to fileURLToPath and appended with file gives us this: `/Users/courey/dev/architect-plugin-search/index.js/launchSearch.js`

import.meta.url passed to path.dirname(): `file:///Users/courey/dev/architect-plugin-search`
```Error: Cannot find module '/Users/courey/dev/gcn.nasa.gov/file:/Users/courey/dev/architect-plugin-search/launchSearch.js'```

if you pass `import.meta.url` directly into `path.dirname()` you get this:
`file:///Users/courey/dev/architect-plugin-search`

If I skipped all that and went straight to using `import.meta.dirname` (which should exist in nodejs 20.11) instead it is undefined, which would give us:
`Error: Cannot find module '/Users/courey/dev/gcn.nasa.gov/undefined/launchSearch.js'`

Reply:
Is it undefined because you are doing a CommonJS build rather than an ESM build?

Reply:
for posterity in case someone gets here and doesn't read all the other comments, we answered this in another comment. It is not supported prior to node v.20.11.0, so it is undefined in any version earlier than that.